src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,28 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
```

Comment:
Verify that `filterSlice` contains the `setFilter`, `setQuery`, and `clearQuery` actions. If any of these actions are missing, it will cause errors when dispatching actions.